server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -59,13 +59,28 @@
 import static java.util.Map.entry;
 import static org.elasticsearch.core.Strings.format;
 
+/**
+ * Manages all the Java thread pools we create. {@link Names} contains a list of the thread pools, but plugins can dynamically add more
+ * thread pools to instantiate.
+ */
 public class ThreadPool implements ReportingService<ThreadPoolInfo>, Scheduler {
 
     private static final Logger logger = LogManager.getLogger(ThreadPool.class);
 
     public static class Names {
+        /**
+         * Often used for quick operations not worth the costs of switching to another thread. Also used to avoid the cost of re-queuing
```

Comment:
The javadoc here should explain what exactly this does (it's certainly not intuitive)

Reply:
Is what you're looking for a prefix to the comment like, "Specifies that the task being scheduled should run on the calling thread, not actually a different thread pool"?

Reply:
Exactly, its not a thread-pool per se

Reply:
Cool, updated the text 👍

Reply:
We're pretty close to being able to drop this constant entirely btw, see https://github.com/elastic/elasticsearch/pull/106279 and other PRs linked from there.

Reply:
https://github.com/elastic/elasticsearch/pull/107249 removes this now-unused symbol.